conans/client/conan_api.py

```diff
@@ -676,16 +685,19 @@ def remote_list_ref(self):
 
     @api_method
     def remote_add_ref(self, reference, remote):
+        reference = ConanFileReference.loads(str(reference))
```

Comment:
The registry now it kind of smart to only associate references without revisions, so I ensure we pass ConanFileReferences always.